pkg/api/v1/defaults.go

```diff
@@ -101,6 +101,14 @@ func SetDefaults_Service(obj *v1.Service) {
 	if obj.Spec.SessionAffinity == "" {
 		obj.Spec.SessionAffinity = v1.ServiceAffinityNone
```

Comment:
also check for ServiceAffinityNone and set `SessionAffinityConfig == nil` ?

Reply:
Currently, we will ignore `SessionAffinityConfig` when ServiceAffinityNone, but I am fine to check it :)